simpeg/electromagnetics/static/resistivity/survey.py

```diff
@@ -24,21 +24,21 @@ class Survey(BaseSurvey):
     def __init__(
         self,
         source_list,
-        survey_geometry="surface",
         **kwargs,
     ):
-        if (key := "survey_type") in kwargs:
-            warnings.warn(
-                f"Argument '{key}' is ignored and will be removed in future "
-                "versions of SimPEG. Types of sources and their corresponding "
-                "receivers are obtained from their respective classes, without "
+        if kwargs.pop("survey_type", None) is not None:
+            raise TypeError(
+                "Argument 'survey_type' has been removed in SimPEG 0.24.0. Types of sources and"
+                "their corresponding receivers are obtained from their respective classes, without "
                 "the need to specify the survey type.",
+            )
+        if kwargs.pop("survey_geometry", None) is not None:
+            warnings.warn(
+                "Argument 'survey_geometry' is deprecated and will be removed in SimPEG 0.25.0.",
                 FutureWarning,
-                stacklevel=0,
+                stacklevel=2,
             )
```

Comment:
@jcapriot, I'm confused why are we deprecating `survey_geometry` here. Would it be possible to do it in a separate PR, so we keep this one with just removals?

Reply:
I'm not sure how this got added in... It shouldn't be here, probably just a late night.

Reply:
Ohh, I see! It's still is. I'll remove that comment then.